### PR TITLE
chore: rename notice to improve clarity

### LINKED ISSUE #770: decreasing_shape_distance and decreasing_stop_time_distance also cover equality cases
**Describe the bug**
Rules decreasing_shape_distance and decreasing_stop_time_distance also cover cases where the shape distance is equal across two records. 

We should reconsider renaming these to make that clearer, as IMHO right now it's a bit confusing when you look at the output and see these notices and then see that the values are equal. It's not clear if it's a bug in the validator or intended behavior.

For example:

```json
{
	"code": "decreasing_stop_time_distance",
	"severity": "ERROR",
	"totalNotices": 2369,
	"notices": [
		{
			"tripId": "120998823",
			"csvRowNumber": 9385505,
			"shapeDistTraveled": 58198.0,
			"stopSequence": 14,
			"prevCsvRowNumber": 7871586,
			"prevStopTimeDistTraveled": 58198.0,
			"prevStopSequence": 13
```

Maybe:
* decreasing_or_equal_shape_distance
* nonincreasing_shape_distance

**How we reproduce the bug**
Steps to reproduce the behaviour:
1. Run on a feed that has sequential equal shape distance or stop time distance records

**Expected behaviour**
Error message should indicate that equality case is covered too

**Observed behaviour**
Error message only indicates that it covers the decreasing case

**Environment versions**
- validator version: master branch at https://github.com/MobilityData/gtfs-validator/commit/dc33783aac7536b17acb3a177f43ebbea2f204eb
- Java version: JDK 11
- OS versions: Windows 10

### PR DESCRIPTION
**Pull request by lionel-nj**
closes #770 

**Summary:**

This PR improves code clarity by renaming two notices: 
- DecreasingShapeDistanceNotice -> DecreasingOrEqualShapeDistanceNotice
- DecreasingStopTimeDistanceNotice -> DecreasingOrEqualStopTimeDistanceNotice

Documentation and validator's javadocs have been updates to match these changes.

**Expected behavior:** 

No code change, same tests should pass. 

Running this branch with the following configuration leads to the following validation report sample: `-u http://webapps.thebus.org/transitdata/Production/google_transit.zip -o output -t 4 -f us-hawai`
Capture d’écran, le 2021-02-24 à 11 49 33

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: RULES.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ Notices are split into three categories: `INFO`, `WARNING`, `ERROR`.
...
 | [`StopTimeWithArrivalBeforePreviousDepartureTimeNotice`](#StopTimeWithArrivalBeforePreviousDepartureTimeNotice)                         	| [E049](https://github.com/MobilityData/gtfs-validator/blob/v1.4.0/RULES.md#E049)      	| Backwards time travel between stops in `stop_times.txt`               	|
 | [`StopTimeWithDepartureBeforeArrivalTimeNotice`](#StopTimeWithDepartureBeforeArrivalTimeNotice), [`StartAndEndTimeOutOfOrderNotice`](#StartAndEndTimeOutOfOrderNotice)                        	| [E045](https://github.com/MobilityData/gtfs-validator/blob/v1.4.0/RULES.md#E045)      	| `arrival_time` after `departure_time` in `stop_times.txt`. `end_time` after `start_time` in `frequencies.txt`.                	|
-| [`DecreasingShapeDistanceNotice`](#DecreasingShapeDistanceNotice)| [E058](https://github.com/MobilityData/gtfs-validator/blob/v1.4.0/RULES.md#E058)   	| Decreasing `shape_dist_traveled` in `stop_times.txt`                    	|
+| [`DecreasingOrEqualShapeDistanceNotice`](#DecreasingOrEqualShapeDistanceNotice)| [E058](https://github.com/MobilityData/gtfs-validator/blob/v1.4.0/RULES.md#E058)   	| Decreasing `shape_dist_traveled` in `stop_times.txt`                    	|
```
Comment (barbeau) [inline RULES.md:43]: The rule description here in the table should also be updated to include equality. Same with the other rule.

**Review by barbeau — CHANGES_REQUESTED** [review]
Thanks @lionel-nj! One small documentation change needed.

File: RULES.md (hunk 1 above)
Comment (barbeau) [reply RULES.md:43]: These descriptions still need to be updated in the table..

**Review by barbeau — CHANGES_REQUESTED** [review]
The table descriptions for the rules still need to be updated.

**Comment by lionel-nj** [comment]
> The table descriptions for the rules still need to be updated.

My bad. Done :)

**Review by barbeau — APPROVED** [review]
LGTM, thanks!

**Pull request merged**